tests/virtualization/xen/save_and_restore.pm

```diff
@@ -37,6 +38,16 @@ sub run {
 
     record_info "Check", "Check restored states";
     assert_script_run "ssh root\@$hypervisor 'virsh list --all | grep $_ | grep running'" foreach (keys %xen::guests);
+
+    record_info "SSH", "Check hosts are listening on SSH";
+    foreach my $guest (keys %xen::guests) {
+        for (my $i = 1; $i <= 60; $i++) {
```

Comment:
If you want you can use `script_retry` from lib/caasp.pm. I plan to export it to utils soon.
```perl
use caasp 'script_retry';
script_retry "ssh root\@$hypervisor 'nmap $guest.$domain -PN -p ssh | grep open'", delay => 60, retry => 60;